### DISCUSSION TITLE
On clustering for pose estimation

### DISCUSSION THREAD

**Discussion by hidmic** [Ekumen-OS/beluga#172] [Ideas]
Connected to #85. Belief tracking with particle filters may run into a multi-modal distribution during operation. `beluga` currently supports weighted mean and covariance statistics for pose estimation, and so it exhibits biased means (w.r.t. to the particles that are actually  tracked) and inflated covariances in such conditions. `nav2_amcl`, on the other hand, works with clusters to reduce mean bias but also artificially reduces covariance.

Should `beluga_amcl` support `nav2_amcl` like pose estimation for feature parity? What would be the correct approach to pose estimation in the face of multi-modal distributions?

**Comment by glpuga** [comment]
I agree this needs to be on the roadmap, but I don't think it's necessarily high priority between now and June.

**Comment by nahueespinosa** [comment]
Closing the discussion as we're already working on this: #275.
